site-src/guides/multiple-ns.md

```diff
@@ -1,38 +1,65 @@
-# Routes in multiple namespaces
-
-With the Gateway API, a single Gateway can target routes across multiple
-namespaces.
-
-This guide assumes that you have installed Gateway API CRDs and a conformant
-controller.
-
-In the following example:
-
-- `acme-lb` GatewayClass: The GatewayClass responsible for satisfying Gateway
-  and Route resources.
-- `multi-ns-gateaway` Gateway: The Gateway is configured with a single listener
-  on port 80 which selects routes that have the label `product: baz` in any
-  namespace.  Notice how the `routes.namespaces.from` field in the listener is
-  set to `All`.
-- `gateway-api-example-ns1` and `gateway-api-example-ns2` Namespaces: These
-  are the namespaces in which route resources are instantiated.
-- `http-app-1` and `http-app-2` HTTPRoutes: These are two resources that are
-  installed in separate namespaces. These routes will be bound to Gateway
-  `multi-ns-gateway` for the following reasons:
-    - Both have the `product: baz` label on them.
-    - `http-app-1` HTTPRoute has `spec.gateways.allow` set to `All`.  The route
-      owner has opted to allow **all** Gateways in the cluster to bind to this
-      Route.
-    - `http-app-2` HTTPRoute has `spec.gateways.allow` set to `FromList` and
-      contains a reference to the `multi-ns-gateway` in `default` namespace.
-      This means that only the specified Gateway resource can bind to this
-      route.  Additional Gateways may be added to this list to allow them to
-      bind to this route.
+# Cross-Namespace routing
 
+The Gateway API is designed with the ability to route across Namespace boundaries. This is useful when there is more than one user or team sharing the same underlying networking infrastructure. Gateways and Routes can be deployed into different Namespaces and select each other across Namespace boundaries. This allows differing user access and roles (RBAC) to be applied to different Namespaces, effectively controlling which parts of the routing configuration are controlled by which teams. This example shows how two teams, who have no direct coordination, can bind to the same Gateway independently from different Namespaces. The ability for Routes to use Gateways across Namespace boundaries is goverend by [_Route binding_](/guides/multiple-ns#cross-namespace-route-binding).
+
+In this example, there are two independent teams, store and site, operating in the same Kubernetes cluster, in the `store` and `site` Namespaces. These are their requirements:
+
+- The site team has two applications, home and login, that are running behind `foo.example.com`. They have decided to deploy independent HTTPRoutes for each application so that the routing can be configured indendently for each app. By having an independent HTTPRoute for each app, it allows them to update the canary weights for the login app without having to touch any routing configuration for home.
```

Comment:
Delegating based on domain is also very common (I would say more common, but I have no evidence :slightly_smiling_face: ), which I think deserves a mention. https://github.com/istio/istio/blob/58867c37948598b2a03e23fa451a8211d7dad18b/pilot/pkg/config/kube/gateway/testdata/delegated.yaml#L24 is an example

Reply:
Separating app components of a public URL I'd say is probably 50/50 by domain/path. I'll make a mention of separating by domain. I think in the future we probably need a separate guide to focus on the actual delegation of parts of the URL in a more secure manner. In this example, I'm assuming all actors have good intent and are somewhat coordinated.

When Route inclusion comes, we will have 2 ways of delegating domain ownership - through Route inclusion and also through Gateway listeners. I think it's a bit early to document good patterns and best practices about these things at this time.